generator/integration_tests/tests/golden_thing_admin_connection_test.cc

```diff
@@ -242,6 +243,44 @@ TEST(GoldenThingAdminConnectionTest, CreateDatabaseCancel) {
   EXPECT_THAT(db, StatusIs(StatusCode::kCancelled));
 }
 
+TEST(GoldenThingAdminConnectionTest, CreateDatabaseStartAwait) {
+  auto mock = std::make_shared<MockGoldenThingAdminStub>();
+  google::longrunning::Operation expected_operation;
+  expected_operation.set_name("test-operation-name");
+  google::test::admin::database::v1::CreateDatabaseMetadata metadata;
+  expected_operation.mutable_metadata()->PackFrom(metadata);
+
+  EXPECT_CALL(*mock, CreateDatabase(_, _, _)).WillOnce([&] {
+    return make_status_or(expected_operation);
+  });
+  EXPECT_CALL(*mock, AsyncGetOperation)
+      .WillOnce([expected_operation](
+                    CompletionQueue&, auto, auto,
+                    google::longrunning::GetOperationRequest const& r) {
+        EXPECT_EQ(expected_operation.name(), r.name());
+        google::longrunning::Operation op;
+        op.set_name(r.name());
+        op.set_done(true);
```

Comment:
comment for myself: I was wondering if we need/want tests for the polling loop side of things? Like, when the operation is not immediately done.

I think no. I think those cases are covered in `async_long_running_operation_test.cc`. We do not need to recreate the tests for `AsyncAwaitLongRunningOperation` in the connections.